### PR TITLE
Fix final size detection for chunked uploads (dav v2 only)

### LINKED ISSUE #179: File size limit is not obeyed when using chunking
files of any size get checked regardless of the "File size limit" setting when chunking is used

### PR DESCRIPTION
**Pull request by VicDeo**
Fixes #179 
Fixes https://github.com/owncloud/enterprise/issues/2145

1. set "File size limit" to 100bytes 
2. upload 3 chunks 40 bytes per chunk
```
user="USER"
pwd="PASSWORD"
host="HOST" # e.g. localhost
path="WEBPATH" # e.g. owncloud
chunk=1234

curl --user $user:$pwd http://$host/$path/remote.php/dav/uploads/$user/$chunk/ --request MKCOL
curl --user $user:$pwd http://$host/$path/remote.php/dav/uploads/$user/$chunk/1 --request PUT --data-binary "@eicar_com.txt_01"
curl --user $user:$pwd http://$host/$path/remote.php/dav/uploads/$user/$chunk/2 --request PUT --data-binary "@eicar_com.txt_02"
curl --user $user:$pwd http://$host/$path/remote.php/dav/uploads/$user/$chunk/3 --request PUT --data-binary "@eicar_com.txt_03"
curl --user $user:$pwd http://$host/$path/remote.php/dav/uploads/$user/$chunk/.file --request MOVE -H "Destination: http://$host/$path/remote.php/dav/files/$user/virus4"
```

### Expected 
`MOVE` request doesn't trigger scanning as 120 bytes are greater than 100
`{"reqId":"ZRjpwP2PYHuAx4Ld9eRi","level":0,"time":"2018-05-21T13:46:58+00:00","remoteAddr":"127.0.0.1","user":"D","app":"files_antivirus","method":"MOVE","url":"\/~deo\/oc-tmp\/remote.php\/dav\/uploads\/D\/117\/.file","message":"File size is 191. av_max_file_size is 100. Scanning is skipped."}
`

### Actual 
it does as file size is considered to be 0:
`{"reqId":"rLxp6i1qRUyv4yYZREhS","level":0,"time":"2018-05-21T14:32:46+00:00","remoteAddr":"127.0.0.1","user":"D","app":"files_antivirus","method":"MOVE","url":"\/~deo\/oc-tmp\/remote.php\/dav\/uploads\/D\/117\/.file","message":"File size is 0. av_max_file_size is 100. Scanning is scheduled."}`

### CODE REVIEW

File: lib/RequestHelper.php [hunk 1]
```diff
@@ -13,52 +13,106 @@
...
+			$cleanedPath
+		);
+		$cachedSize = $this->getCache()->get($translatedPath);
+		if ($cachedSize > 0) {
+			return $cachedSize;
+		}
 
 		$requestMethod = $this->request->getMethod();
+		// Are we uploading anything?
+		if ($requestMethod !== 'PUT') {
```
Comment (PVince81) [inline lib/RequestHelper.php]: maybe put this before all the regexp logic ?

File: lib/RequestHelper.php [hunk 2]
```diff
@@ -13,52 +13,106 @@
...
-
-			if ($requestMethod === 'PUT') {
-				$uploadSize = (int)$this->request->getHeader('CONTENT_LENGTH');
-			} else {
-				$uploadSize = (int)$this->request->getHeader('OC_TOTAL_LENGTH');
-			}
-		} else if ($requestMethod === 'PUT' && $isPublicScript) {
-			$uploadSize = (int)$this->request->getHeader('CONTENT_LENGTH');
 		}
+		$uploadSize = (int)$this->request->getHeader('CONTENT_LENGTH');
```
Comment (PVince81) [inline lib/RequestHelper.php]: are we not using OC_TOTAL_LENGTH for the final MOVE in new chunking ?

Comment (VicDeo) [reply lib/RequestHelper.php]: @PVince81 it's used but could be omited https://github.com/owncloud/core/blob/master/apps/dav/lib/Upload/ChunkingPlugin.php#L97

File: lib/RequestHelper.php (hunk 1 above)
Comment (VicDeo) [reply lib/RequestHelper.php]: mhm. No, but I'd better wrap L75-90 with `if ($requestMethod !== 'MOVE') {`

File: lib/RequestHelper.php (hunk 2 above)
Comment (VicDeo) [reply lib/RequestHelper.php]: Well, I answered your question correctly, but neither the question not the answer is applicable here. This part is for non-chunked upload as the new chunking is handled with a Dav plugin above.

**Comment by PVince81** [comment]
@individual-it can you validate this ?

**Pull request merged**